Automattic/wp-calypso
# GS on Personal A/B: Update the theme details in Pattern Assembler and Theme Showcase

Related to https://github.com/Automattic/dotcom-forge/issues/2858

## Proposed Changes

This PR primarily implements an A/B test, calypso_global_styles_personal, to evaluate user reactions and actions based on different plan offerings associated with Custom Styles. The A/B test presents users with one of two descriptions, depending on the variation, with each description referring to a different plan type - Personal or Premium.

As a secondary change, this PR refactors the term "Premium Styles" to "Custom Styles" throughout the relevant codebase. 


| Before | After |
-----|-----
![miguel-original jgp](https://github.com/Automattic/wp-calypso/assets/1989914/fee7d6e1-d0ac-4872-b66c-b96cc90f2a07)|![Captura de Pantalla 2023-07-06 a las 14 02 12](https://github.com/Automattic/wp-calypso/assets/1989914/b8016926-c342-4355-8f19-88a1434f7d7f)


## Testing Instructions



* Start Calypso.
* Navigate to the following URLs and ensure the changes are visible:
  * http://calypso.localhost:3000/theme/storia/{yoursite}?style_variation=green
  * http://calypso.localhost:3000/setup/site-setup/designSetup?siteSlug={your site}&ref=site-selector&theme=bitacora&style_variation=antigua
* Confirm the change from "Premium Styles" to "Custom Styles".
* Verify that the description changes to refer to the Personal plan if the A/B test variation is set to "treatment". For this, you'll need to manually set the A/B test variation to "treatment".
* Also, check that the description refers to the Premium plan when the A/B test variation is not set to "control".

You can change your A/B group [here](21268-explat-experiment).
When changing groups make sure to purge the cache with `wp_cache_delete( 'global-styles-on-personal-{sideId}', 'a8c_experiments' );`

## Pre-merge Checklist
